src/elements/paginator/paginator.ts

```diff
@@ -0,0 +1,345 @@
+
+import style from './paginator.scss?lit&inline';
+
+import '../button/mini-button.js';
+import '../button/mini-button-group.js';
+import '../divider.js';
+import '../form-field.js';
+import '../select.js';
+import '../option.js';
+
+export type SbbPaginatorPageChanged = {
+  currentPageIndex: number;
+  previousPageIndex: number;
+};
+
+const MAX_PAGE_NUMBERS_DISPLAYED = 3;
+
+/**
+ * It displays a paginator component.
+ *
+ * @event {CustomEvent<SbbPaginatorPageChanged>} pageChanged - Emits when the pageIndex changes.
+ */
+@customElement('sbb-paginator')
+export class SbbPaginatorElement extends SbbNegativeMixin(LitElement) {
+  public static override styles: CSSResultGroup = style;
+  public static readonly events: Record<string, string> = {
+    pageChanged: 'pageChanged',
+  } as const;
+
+  /** Total number of items. */
+  @property({ type: Number })
+  public set length(value: number) {
+    this._length = isNaN(value) ? 0 : Math.max(value, 0);
+  }
+  public get length(): number {
+    return this._length;
+  }
+  private _length: number = 0;
+
+  /** Number of items per page. */
+  @property({ attribute: 'page-size', type: Number })
+  public set pageSize(value: number) {
+    const previousPageSize = this._pageSize;
+    this._pageSize = Math.max(value, 0);
+    this.pageIndex = Math.floor((this.pageIndex * previousPageSize) / this._pageSize) || 0;
+  }
+  public get pageSize(): number {
+    return this._pageSize;
+  }
+  private _pageSize: number = 10;
+
+  /** Current page index. */
+  @property({ attribute: 'page-index', type: Number })
+  public set pageIndex(value: number) {
+    const previousPageIndex = this._pageIndex;
+    this._pageIndex = this._validatePageIndex(value);
+    if (previousPageIndex !== this._pageIndex) {
+      this._pageChanged.emit({ previousPageIndex, currentPageIndex: value });
+    }
+  }
+  public get pageIndex(): number {
+    return this._pageIndex;
+  }
+  private _pageIndex: number = 0;
+
+  /** The available `pageSize` choices. */
+  @property({ attribute: 'page-size-options', type: Array }) public pageSizeOptions?: number[];
+
+  /** Position of the prev/next buttons. */
+  @property({ attribute: 'pager-position' }) public pagerPosition: 'start' | 'end' = 'start';
+
+  /** Size variant, either m or s. */
+  @property({ reflect: true }) public size: 'm' | 's' = 'm';
+
+  private _pageChanged: EventEmitter<SbbPaginatorPageChanged> = new EventEmitter(
+    this,
+    SbbPaginatorElement.events.pageChanged,
+    { composed: true, bubbles: true },
+  );
+
+  private _abort = new SbbConnectedAbortController(this);
+  private _language = new SbbLanguageController(this);
+  private _markForFocus: number | null = null;
+
+  protected override firstUpdated(changedProperties: PropertyValues<this>): void {
+    super.firstUpdated(changedProperties);
+    const signal = this._abort.signal;
+    this.shadowRoot!.querySelector('.sbb-paginator__pages')!.addEventListener(
```

Comment:
i have to remove this, since as per Mürsel review the arrow navigation is out of standard and it has to be replaced with tab navigation as in the sbb-angular